app/src/org/commcare/network/HttpUtils.java

```diff
@@ -3,15 +3,42 @@
 import android.content.SharedPreferences;
 
 import org.commcare.CommCareApplication;
+import org.commcare.core.network.CommCareNetworkService;
+import org.commcare.core.network.CommCareNetworkServiceGenerator;
+import org.commcare.core.network.ModernHttpRequester;
 import org.commcare.modern.util.Pair;
+import org.commcare.preferences.CommCarePreferences;
+import org.commcare.preferences.DeveloperPreferences;
+import org.commcare.utils.CredentialUtil;
 import org.commcare.utils.SessionUnavailableException;
 import org.javarosa.core.model.User;
 
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import retrofit2.Response;
+
 /**
  * @author Phillip Mates (pmates@dimagi.com)
  */
 public class HttpUtils {
-    public static Pair<User, String> getUserAndDomain(boolean isAuthenticatedRequest) {
+
+    public static String getCredential(@Nullable Pair<String, String> usernameAndPasswordToAuthWith) {
+        String credential;
+        if (usernameAndPasswordToAuthWith == null) {
```

Comment:
Ah gotcha, makes sense. It looks like that method hasn't made its way into this class though, btw